tests/test_proxy_functional.py

```diff
@@ -3,13 +3,52 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
+import trustme
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def trustme_certs(tmp_path):
+    ca = trustme.CA()
+    ca.issue_cert("localhost")
+    ca_cert_path = tmp_path / "ca.pem"
+    ca.cert_pem.write_to_path(ca_cert_path)
+    ca_key_path = tmp_path / "server.key"
+    ca.private_key_pem.write_to_path(ca_key_path)
+    return (str(ca_cert_path), str(ca_key_path))
+
+
+@pytest.fixture
+def secure_proxy_url(trustme_certs):
+    port = proxy.common.utils.get_available_port()
```

Comment:
Even if you don't end up relying on an ephemeral port, you could just use the `aiohttp_unused_port` fixture — no need to use a helper from `proxy.py` because it's not a public API (I suppose) and may break over time.